library/core/src/sync/atomic.rs

```diff
@@ -914,6 +936,30 @@ impl<T> AtomicPtr<T> {
         unsafe { &mut *self.p.get() }
     }
 
+    /// Get atomic access to a pointer.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(atomic_from_mut)]
+    /// use std::sync::atomic::{AtomicPtr, Ordering};
+    ///
+    /// let mut some_ptr = &mut 123 as *mut i32;
+    /// let a = AtomicPtr::from_mut(&mut some_ptr);
+    /// a.store(&mut 456, Ordering::Relaxed);
+    /// assert_eq!(unsafe { *some_ptr }, 456);
+    /// ```
+    #[inline]
+    #[unstable(feature = "atomic_from_mut", issue = "none")]
+    pub fn from_mut(v: &mut *mut T) -> &Self {
```

Comment:
This one seems a little strange to me, since `*mut T` is already a pointer.

Reply:
Just like `AtomicU16::from_mut` will take a `&mut u16`, `AtomicPtr<T>::from_mut` should take a `&mut` to the non-atomic type: `&mut *mut T`. The already existing `AtomicPtr<T>::get_mut()` also returns a `&mut *mut T`: <https://doc.rust-lang.org/stable/core/sync/atomic/struct.AtomicPtr.html#method.get_mut>

Reply:
@KodrAus The pointer itself is what atomic here, not what it points to, so when using is you actually use a pointer to the atomic pointer (otherwise you'll be copying the atomic pointer)